src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -33,14 +35,18 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(Authentication authentication) {
+        User user = userService.findByEmail(authentication.getName()).orElseThrow(
+                () -> new NoSuchElementException("Invalid email"));
```

Comment:
```suggestion
                () -> new NoSuchElementException("Invalid email: " + authentication.getName()));
```
the same for other similar cases